pages/posts/share-contact-and-send-message.tsx

```diff
@@ -0,0 +1,452 @@
+                </p>
+                <h3>Helper functions</h3>
+                <p>
+                  That’s it for main! Now we need to define the helper functions <code>receiveContactRequest</code> and <code>receiveMessage</code>. These both follow the pattern
+                  of subscribing to an event stream and waiting for the desired event type. Paste the following function to the file <code>contact.go</code>.
+                </p>
+                <pre>
+                  <code className="language-go">
+                    {`
+	func receiveContactRequest(ctx context.Context, client weshnet.ServiceClient) (*protocoltypes.AccountContactRequestIncomingReceived, error) {
+		// Get the client's AccountGroupPK from the configuration.
+		config, err := client.ServiceGetConfiguration(ctx,
+			&protocoltypes.ServiceGetConfiguration_Request{})
+		if err != nil {
+			return nil, err
+		}
+
+		// Subscribe to metadata events. ("sub" means "subscription".)
+		subCtx, subCancel := context.WithCancel(ctx)
+		defer subCancel()
+		subMetadata, err := client.GroupMetadataList(subCtx,
+			&protocoltypes.GroupMetadataList_Request{
+				GroupPK: config.AccountGroupPK,
+			})
+		if err != nil {
+			return nil, err
+		}
+
+		for {
+			metadata, err := subMetadata.Recv()
+			if err == io.EOF || subMetadata.Context().Err() != nil {
+				// Not received.
+				return nil, nil
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			if metadata == nil || metadata.Metadata.EventType !=
+					protocoltypes.EventTypeAccountContactRequestIncomingReceived {
+				continue
+			}
+
+			request := &protocoltypes.AccountContactRequestIncomingReceived{}
+			if err = request.Unmarshal(metadata.Event); err != nil {
+				return nil, err
+			}
+
+			return request, nil
+		}
+	}
+`}
+                  </code>
+                </pre>
+                <p>
+                  This function takes the client1 which we created in <code>main</code> and calls <code>ServiceGetConfiguration</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.ServiceGetConfiguration">docs</a>). Instead of getting
+                  the configuration’s peer ID as in previous examples, we get the public key of client1’s Account group. This public key is also in the shared contact, and is used
+                  by client2 to send a contact request to client1. To receive it, we call <code>GroupMetadataList</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.GroupMetadataList">docs</a>) with the Account group
+                  public key.
+                </p>
+                <p>
+                  Most API functions return a data structure, but a few like <code>GroupMetadataList</code> return a subscription stream like <code>subMetadata</code>. We use a
+                  &nbsp;<code>for</code> loop and call
```

Comment:
As mentioned by @LFGaming , there are two spaces before "for". Maybe remove the `&nbsp;` ?